payments-core/src/main/java/com/stripe/android/payments/paymentlauncher/PaymentLauncherViewModel.kt

```diff
@@ -1,32 +1,67 @@
 package com.stripe.android.payments.paymentlauncher
 
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
+import com.stripe.android.StripeIntentResult
+import com.stripe.android.exception.APIException
 import com.stripe.android.model.ConfirmStripeIntentParams
+import com.stripe.android.model.StripeIntent
 
 /**
  * WIP - view model for PaymentLauncherHostActivity
  */
+
 internal class PaymentLauncherViewModel : ViewModel() {
+    /**
+     * [PaymentResult] live data to be observed.
+     */
+    internal val paymentLauncherResult = MutableLiveData<PaymentResult>()
 
     /**
      * Confirms a payment intent or setup intent
      */
-    fun confirmStripeIntent(confirmStripeIntentParams: ConfirmStripeIntentParams) {}
+    internal fun confirmStripeIntent(confirmStripeIntentParams: ConfirmStripeIntentParams) {}
 
     /**
-     * Fetches a payment intent and handles its next action.
+     * Fetches a [StripeIntent] and handles its next action.
      */
-    fun handleNextActionForPaymentIntent(clientSecret: String) {}
+    internal fun handleNextActionForStripeIntent(clientSecret: String) {}
 
     /**
-     * Fetches a setup intent and handles its next action.
+     * Parse [StripeIntentResult] into [PaymentResult].
      */
-    fun handleNextActionForSetupIntent(clientSecret: String) {}
+    private fun postResult(stripeIntentResult: StripeIntentResult<StripeIntent>) {
```

Comment:
Is this an intentional change, just want to make sure I understand, seems different from the focus of the review.

Reply:
Already synced